core/src/main/scala/codesearch/core/index/directory/PackageDirectory.scala

```diff
@@ -0,0 +1,49 @@
+package codesearch.core.index.directory
+
+import java.nio.file.{Path, Paths}
+
+import codesearch.core.index.directory.PathOps._
+import codesearch.core.index.repository._
+
+object PathOps {
+  implicit final class RichPath(val parent: Path) {
+    def /(child: Path): Path   = Paths.get(s"${parent.toFile.getPath}/${child.toFile.getPath}")
+    def /(child: String): Path = Paths.get(s"${parent.toFile.getPath}/$child")
+  }
+}
+
+trait Directory[A <: SourcePackage] {
+  def root: Path        = Paths.get(s"./data")
+  def extension: String = "tgz"
+  def archive(pack: A): Path
+  def unarchived(pack: A): Path = archive(pack).getParent
+}
+
+object PackageDirectory {
+
+  implicit class PackageDirectoryOps[A <: SourcePackage](val pack: A) {
+    def archive(implicit ev: Directory[A]): Path    = ev.archive(pack)
+    def unarchived(implicit ev: Directory[A]): Path = ev.unarchived(pack)
+  }
+
+  implicit def hackageDirectory: Directory[HackagePackage] = new Directory[HackagePackage] {
+    override def archive(pack: HackagePackage): Path =
+      root / "hackage" / s"${pack.name}" / s"${pack.version}" / s"${pack.name}-${pack.version}.$extension"
```

Comment:
Why does it say `root` here? I guess we want relative paths, or I'm not sure what we want – but definitely not putting everything into the server root

Reply:
`root` value defined in trait Directory by default `./data`

Reply:
alright, but then let's rename it to e.g. `dataRoot` or something, because otherwise it can be mixed up with `Ammonite.root`